### PR TITLE
AffineTransform: Align with Darwin and misc fixes

### PR DESCRIPTION
**Pull request by spevans**
- init(): Return a matrix with identity values.

- NSAffineTransform: Fix constructor
  convenience init(transform: NSAffineTransform) ->
  convenience init(transform: AffineTransform)

- Add NSAffineTransformStruct type and use for
  NSAffineTransform.transformStruct

- NSAffineTransform: Store a private AffineTransform to use as
  backing.

- Fixup tests.

### CODE REVIEW

**Comment by spevans** [comment]
@swift-ci please test

**Comment by ianpartridge** [comment]
This looks good to me, but as it adds `NSAffineTransformStruct` I'll let @parkera confirm there's no problems I'm unaware of.

File: Foundation/AffineTransform.swift [hunk 1]
```diff
@@ -279,8 +273,57 @@ public struct AffineTransform : ReferenceConvertible, Hashable, CustomStringConv
...
 
+public struct NSAffineTransformStruct {
+    public var m11: CGFloat
+    public var m12: CGFloat
+    public var m21: CGFloat
+    public var m22: CGFloat
+    public var tX: CGFloat
+    public var tY: CGFloat
+
+    // This is NOT an identity matrix
```
Comment (parkera) [inline Foundation/AffineTransform.swift]: Maybe this is worth putting as an actual documentation comment on this thing.

**Comment by spevans** [comment]
@parkera I documented all the methods, is the documentation format correct?

**Comment by parkera** [comment]
That's great, thanks!

**Comment by ianpartridge** [comment]
@swift-ci please test and merge

**Comment by ianpartridge** [comment]
@swift-ci please test and merge

**Pull request merged**